apps/vs-code-designer/src/app/commands/funcCoreTools/validateFuncCoreToolsIsLatest.ts

```diff
@@ -2,132 +2,60 @@
-import type { FuncVersion } from '@microsoft/vscode-extension';
 import * as semver from 'semver';
 import type { MessageItem } from 'vscode';
 
-export async function validateFuncCoreToolsIsLatest(): Promise<void> {
+export async function validateFuncCoreToolsIsLatest(majorVersion?: string): Promise<void> {
   await callWithTelemetryAndErrorHandling('azureLogicAppsStandard.validateFuncCoreToolsIsLatest', async (context: IActionContext) => {
     context.errorHandling.suppressDisplay = true;
     context.telemetry.properties.isActivationEvent = 'true';
 
-    const showMultiCoreToolsWarningKey = 'showMultiCoreToolsWarning';
-    const showMultiCoreToolsWarning = !!getWorkspaceSetting<boolean>(showMultiCoreToolsWarningKey);
-
     const showCoreToolsWarningKey = 'showCoreToolsWarning';
     const showCoreToolsWarning = !!getWorkspaceSetting<boolean>(showCoreToolsWarningKey);
 
-    if (showCoreToolsWarning || showMultiCoreToolsWarning) {
-      const packageManagers: PackageManager[] = await getFuncPackageManagers(true /* isFuncInstalled */);
-      let packageManager: PackageManager;
-
-      if (packageManagers.length === 0) {
-        return;
-      } else if (packageManagers.length === 1) {
-        packageManager = packageManagers[0];
-        context.telemetry.properties.packageManager = packageManager;
-      } else {
-        context.telemetry.properties.multiFunc = 'true';
-
-        if (showMultiCoreToolsWarning) {
-          const message: string = localize('multipleInstalls', 'Detected multiple installs of the func cli.');
-          const selectUninstall: MessageItem = { title: localize('selectUninstall', 'Select version to uninstall') };
-          const result: MessageItem = await context.ui.showWarningMessage(message, selectUninstall, DialogResponses.dontWarnAgain);
-
-          if (result === selectUninstall) {
-            await executeOnFunctions(uninstallFuncCoreTools, context, packageManagers);
+    const binaries = binariesExist(funcDependencyName);
+    context.telemetry.properties.binariesExist = `${binaries}`;
+
+    if (!binaries) {
+      await installFuncCoreTools(context, majorVersion);
+      context.telemetry.properties.binaryCommand = `${getFunctionsCommand()}`;
+    } else if (showCoreToolsWarning) {
+      context.telemetry.properties.binaryCommand = `${getFunctionsCommand()}`;
+      const localVersion: string | null = await getLocalFuncCoreToolsVersion();
+      context.telemetry.properties.localVersion = localVersion;
+      const newestVersion: string | undefined = await getLatestFunctionCoreToolsVersion(context, majorVersion);
+      if (semver.major(newestVersion) === semver.major(localVersion) && semver.gt(newestVersion, localVersion)) {
+        context.telemetry.properties.outOfDateFunc = 'true';
+        const message: string = localize(
+          'outdatedFunctionRuntime',
+          'Update your Azure Functions Core Tools ({0}) to the latest ({1}) for the best experience.',
```

Comment:
Consistency edit after getting more context:

```suggestion
          'Update your local Azure Functions Core Tools version ({0}) to the latest version ({1}) for the best experience.',
```